### ISSUE TITLE
[$250] Submit Expense - Submit button appears briefly In expense details page

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#53263]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.68-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** biruknew45+5gg@gmail.com
**Issue reported by:** Applause -  Internal Team

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. As an admin, create a workspace and add an employee
3. As an admin, set delay submission to manually
4. As an employee, go to the workspace chat and submit an expense
5. As an employee, click on the submit button and then click on the expense preview quickly

## Expected Result:
Submit button shouldn't appear

## Actual Result:
Submit button appears and then disappears

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/215f68cb-9289-4c62-b08f-7dd1ed90835e

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021864310032102390236</li>
        <li>Upwork Job ID: 1864310032102390236</li>
        <li>Last Price Increase: 2025-01-01</li>
    </ul>
</details>

**Comment by linhvovan29546** [comment]
@ntdiary You are correct, the `SubmitReport` request was already sent to the backend. However, the `GetMissingOnyxMessages` request was called with the parameters `updateIDFrom` and `updateIDTo`, specifying a range that predates the `SubmitReport` call. (I’m not entirely sure I understand the exact logic in the backend for `GetMissingOnyxMessages`.)
<details>
 <summary>Details</summary>
 <img width="1055" alt="Screenshot 2024-12-06 at 7 32 24 PM" src="https://github.com/user-attachments/assets/289552c0-bb6c-425b-871f-5f9930a27e1f">
</details>

**Comment by linhvovan29546** [comment]
I also noticed that the `previousUpdateID` of `SubmitReport` seems unusual. It might be an issue with the backend logic, as you mentioned

> @ntdiary You are correct, the `SubmitReport` request was already sent to the backend. However, the `GetMissingOnyxMessages` request was called with the parameters `updateIDFrom` and `updateIDTo`, specifying a range that predates the `SubmitReport` call. (I’m not entirely sure I understand the exact logic in the backend for `GetMissingOnyxMessages`)
> 
> Details

**Comment by linhvovan29546** [comment]
I also reproduced the issue using different steps without opening the Expense details page. The root cause appears to be that the `SubmitReport` response returns a `previousUpdateID` that differs from the `clientUpdateID` in the request header. As a result, the app detects a conflict and resolves it by making a `GetMissingOnyxMessages` request with a range that predates the `SubmitReport` call.
<details>
 <summary>Details</summary>
<img    width="879"  alt="Screenshot 2024-12-07 at 8 11 03 AM" src="https://github.com/user-attachments/assets/da971b7f-685c-4fa6-98d1-944a0218fbd4">
<img width="879" alt="Screenshot 2024-12-07 at 8 11 21 AM" src="https://github.com/user-attachments/assets/17f24262-e8fe-48cc-9b14-44a5ac5128b4">

</details>

**Comment by linhvovan29546** [comment]
## Proposal
Update the root cause of the problem and include an alternative solution to fix it via the backend. [Updated](https://github.com/Expensify/App/issues/53263#issuecomment-2520255605)

**Comment by FitseTLT** [comment]
yep @ntdiary you are right. Unfortunately they have applied a FE workaround [here](https://github.com/Expensify/App/pull/53541) to fix it which I don't feel it is correct but you might not be able to reproduce it anymore.

**Comment by rafecolton** [comment]
> Maybe @rafecolton has already checked the backend logic, and think adding SUBMIT_REPORT to requestsToIgnoreLastUpdateID is sufficient. :)

I did not check this

**Comment by rafecolton** [comment]
What needs to happen here? Does the FE PR need to be reverted? Do we need to recruit somebody to investigate a BE change the FE change masked?

**Comment by muttmuure** [comment]
It sounds like the latter

**Comment by muttmuure** [comment]
I've added the Internal label and added to Hot Picks

**Comment by rafecolton** [comment]
Thanks Matt!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by muttmuure** [comment]
Waiting for someone internal to pick this up

**Comment by muttmuure** [comment]
No update
